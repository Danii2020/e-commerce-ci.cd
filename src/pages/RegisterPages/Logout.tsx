import { signOut } from "firebase/auth"
import { auth } from "../../lib/firebase/firebaseConfig"

// Using built in signOut method to return user state to default/null
const control = ()   => {

  const onSignOut = () => {
    try {
        signOut(auth);
    } catch (err: any) {
      console.error("Logout error:", err.message);
    }
  }


  return (
      <div>
        <button onClick={onSignOut} >Logging Out!</button>
      </div>
  )
}

export default control