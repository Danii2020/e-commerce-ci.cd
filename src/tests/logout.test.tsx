import { render } from '@testing-library/react'
import Logout from '../pages/RegisterPages/Logout'

test ('matches snapshot', () => {
    const {asFragment} = render(<Logout />);

    expect(asFragment()).toMatchSnapshot();
})