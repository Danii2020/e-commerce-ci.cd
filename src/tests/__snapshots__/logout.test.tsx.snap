// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Logging Out!
    </button>
  </div>
</DocumentFragment>
`;
